src/main/scala/day1/Exercises.scala

```diff
@@ -0,0 +1,96 @@
+package day1
+
+import sun.text.normalizer.Replaceable
+
+object Exercises {
+    def main(args: Array[String]): Unit ={
+        sayThis(helloWorld())
+        printTheInput(5)
+        printTheInput(5.5)
+        printTheInput("cat")
+        println(stringEnder("Thomas", 20))
+        println(stringEnder("Thomas", 4))
+        printTheInput(stringMergeAndReplace("Hal", "lo", 'a','e'))
+        println(addOrMult(2, 3, true))
+        println(addOrMult(2, 3, false))
+        println(addOrMult(0, 0, false))
+        println(addOrMult(5, 0, false))
+        spamOut("You must construct additional pylons", 5)
+        spamSquare("woah", 5)
+        fizzbuzz("fizz", "buzz", 15)
+    }
+    def helloWorld(): String = {
+        val hello = "Hello World"
+        hello
+    }
+    def sayThis(input: String){
+        println(input)
+    }
+    def printTheInput(input: Any) = {
+        println(input)
+    }
+    def stringEnder(word: String, number: Integer): String ={
+        val wordLength = word.length()
+        if(wordLength <= number){
+            word
+        }
+        else{
+            word.substring(wordLength - number)
+        }
+    }
+    def stringMergeAndReplace(firstHalf: String, secondHalf: String, replaceThis: Char, replaceWith: Char): String ={
+        val ourString = (firstHalf + secondHalf).replace(replaceThis, replaceWith)
+        return ourString
+    }
+    def addOrMult(intOne: Integer, intTwo: Integer, doYouAdd: Boolean): Integer = {
+        if(intOne == 0  && intTwo == 0){
+            0
+        }
+        else if(intOne == 0){
+            intTwo
+        }
+        else if(intTwo == 0){
+            intOne
+        }
+        else if(doYouAdd){
+            intOne + intTwo
+        }
+        else {
+            intOne * intTwo
+        }
+    }
+    def spamOut(phrase: String, times: Integer): Unit ={
+            println(phrase)
+            if(times != 0){
+                var times2 = times - 1
+                spamOut(phrase, times2)
```

Comment:
instead of 
var times2 = times - 1
spamOut(phrase, times2)

you could just have
spamOut(phrase, times - 1)